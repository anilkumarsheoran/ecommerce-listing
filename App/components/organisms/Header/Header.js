import PropTypes from "prop-types";
import styles from "./Header.style";
import withStyles from "../../../global/withStyle";
import { heading } from "../../../locales/en-US";
import SelectBox from "../../atoms/SelectBox";
import { filterSizes } from "../../../constants/index";

/**
 *  Header component can be used to display the List of products.
 *  @param {function} filterProducts : is a callback function, which mutate the productData to be displayed;
 *  @param {string} className :  className is string , which is returned by styled components;
 */

const Header = props => {
  const { filterProducts, className } = props;
  return (
    <div className={className}>
      <h2>{heading}</h2>
      <SelectBox
        options={filterSizes}
        onSelect={selected => filterProducts(selected)}
        className="filter"
      />
    </div>
  );
};

Header.propTypes = {
  filterProducts: PropTypes.func.isRequired,
  className: PropTypes.string
};

export default withStyles(Header, styles);
export { Header };
