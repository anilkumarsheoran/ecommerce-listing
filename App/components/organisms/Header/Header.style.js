import { css } from "styled-components";
import theme from "../../../styles/theme";

export default css`
  background-color: ${theme.colors.blue};
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 15px 10px;
  margin: 25px 17px 10px;
  h2 {
    margin: 0;
  }
  .filter select {
    width: 150px;
    font-size: 14px;
  }
`;
