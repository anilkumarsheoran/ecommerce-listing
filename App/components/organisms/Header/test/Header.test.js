import { shallow } from "enzyme";
import { Header } from "../Header";
const { configure } = require("enzyme");
const Adapter = require("enzyme-adapter-react-16");

configure({ adapter: new Adapter() });

const props = {
  filterProducts: jest.fn()
};

describe("Header Component", () => {
  test("should render correctly", () => {
    const header = shallow(<Header {...props} />);
    expect(header).toMatchSnapshot();
  });
});
