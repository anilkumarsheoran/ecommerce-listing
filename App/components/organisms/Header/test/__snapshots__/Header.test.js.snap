// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    filterProducts={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Women's tops
        </h2>,
        <Styled(SelectBox)
          className="filter"
          onSelect={[Function]}
          options={
            Array [
              Object {
                "label": "Filter By Size",
                "value": "",
              },
              Object {
                "label": "XS",
                "value": "XS",
              },
              Object {
                "label": "S",
                "value": "S",
              },
              Object {
                "label": "M",
                "value": "M",
              },
              Object {
                "label": "XL",
                "value": "XL",
              },
              Object {
                "label": "L",
                "value": "L",
              },
            ]
          }
        />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Women's tops",
        },
        "ref": null,
        "rendered": "Women's tops",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "filter",
          "onSelect": [Function],
          "options": Array [
            Object {
              "label": "Filter By Size",
              "value": "",
            },
            Object {
              "label": "XS",
              "value": "XS",
            },
            Object {
              "label": "S",
              "value": "S",
            },
            Object {
              "label": "M",
              "value": "M",
            },
            Object {
              "label": "XL",
              "value": "XL",
            },
            Object {
              "label": "L",
              "value": "L",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Women's tops
          </h2>,
          <Styled(SelectBox)
            className="filter"
            onSelect={[Function]}
            options={
              Array [
                Object {
                  "label": "Filter By Size",
                  "value": "",
                },
                Object {
                  "label": "XS",
                  "value": "XS",
                },
                Object {
                  "label": "S",
                  "value": "S",
                },
                Object {
                  "label": "M",
                  "value": "M",
                },
                Object {
                  "label": "XL",
                  "value": "XL",
                },
                Object {
                  "label": "L",
                  "value": "L",
                },
              ]
            }
          />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Women's tops",
          },
          "ref": null,
          "rendered": "Women's tops",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "filter",
            "onSelect": [Function],
            "options": Array [
              Object {
                "label": "Filter By Size",
                "value": "",
              },
              Object {
                "label": "XS",
                "value": "XS",
              },
              Object {
                "label": "S",
                "value": "S",
              },
              Object {
                "label": "M",
                "value": "M",
              },
              Object {
                "label": "XL",
                "value": "XL",
              },
              Object {
                "label": "L",
                "value": "L",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
